lib/tasks/test_factories_lint.rake

```diff
@@ -1,12 +1,41 @@
 # frozen_string_literal: true
 
+require './spec/support/contract_helper.rb'
+require 'pry'
+
 namespace :test do
   namespace :factories do
     desc 'Lint the factories'
     task lint: :environment do
-      puts 'Linting factories...'
-      FactoryBot.lint
-      puts 'Done'
+      require 'webmock'
+      include WebMock::API
+
+      WebMock.enable!
+      # We currently use mocks in the factories to handle building API
+      # objects. We should look at better ways of handling this, but until
+      # then we enable mocks while linting.
```

Comment:
What are the better ways of handling this?

Reply:
I'm not 100% sure. The comment was initially made before I made the changes which eliminated most of the mocks in the factories, but a few still remain.

One prospect would be to generate json-api output and get the gem to parse this as though it came in via the API directly. This would ensure that our test object behaved exactly like our real ones, but at the expense of added complexity (and probably reduced speed) in the factories.

It still causes a bit of trouble with the ancestor scope object though, where we end up mocking the results of the query. Not sure how to handle this one without mocks. Could use webmock, but that feels a bit like replacing one smell with another. It probably indicates that ancestors don't belong in the factories, and should be set up separately in the tests themselves.

Reply:
I'm not too familiar with all the options, but opinions from what I've encountered so far:

I enjoyed making changes like this - https://github.com/sanger/asset_audits/pull/132/files#diff-0b60e12263cf050d50a27eda10159846 - where the old mocking was v verbose and not v readable. 
Seems like generating json-api output would bring back in that complexity again? Maybe not, though, if we wrote some nice helper methods.

Eduardo said, in a conversation about the mocking for asset audits, 
"yes, better to use that mocha
webmock only if you need it for any reason, but if not, better avoid it
because any change in http metadata will break the mock"

so probs worth having a wider chat with the team about best way to mock api calls?